torch/library.h

```diff
@@ -202,6 +203,46 @@ namespace detail {
 
   class TorchLibraryInit;
 
+} // namespace detail
+
+// Note [Selective build]
+// ~~~~~~~~~~~~~~~~~~~~~~
+// In some settings, especially mobile, it is important to avoid compiling any
+// references to functions that you aren't actually going to use, so that they
+// can be eliminated by the linker.  We call this capability "selective build".
+//
+// A very easy way to implement selective build which results in a lot of
+// boilerplate is to just add ifdef's around every registration call, but this
+// means you have to write a lot of extra lines of code at every registration
+// site, and it also means you have to define some munging scheme to map
+// operators to macros.
+//
+// Instead of doing this, we have a different mechanism centered around the
+// concept of a SelectiveName.  A selective name is like a const char* string,
+// except it also carries at compile time a boolean saying whether or not a
+// registration should actually happen or not.  We then have extra overloads which
+// bypass registration entirely if a selective name is disabled.  We do a
+// constexpr test to see if a operator should be enabled or not; this is
+// currently implemented in ATen/core/op_registration/op_whitelist.h
+
+namespace detail {
+
+  // A SelectiveName is like a const char*, except that it also comes
+  // with a type brand that says whether or not the name is enabled or
+  // not.  If a name is disabled, then (at compile time) we DON'T generate
+  // a registration call for it.  This class is not intended to be called
+  // directly; use TORCH_SELECTIVE_NAME macro below to create it.
+  template <bool enabled>
+  class SelectiveName {
+  public:
+    constexpr SelectiveName(const char* name) : name_(name) {}
+    constexpr operator const char*() { return name_; }
+  private:
+    const char* name_;
```

Comment:
We should use c10::string_view here and also rewrite the constexpr string functions to use it

Reply:
Mm, string view here isn't really necessary. Is there any particular reason you want a string view here?

Reply:
just generally prefer string_view to const char* because it is more flexible and doesn't have downsides. Say at some point you want to instantiate a SelectiveStr holding only the op name from a `const char*` holding the full schema, `string_view` would easily allow that while `const char*` has to be null terminated. But not really important here.